middleware.js

```diff
@@ -68,6 +68,9 @@ function middleware (api, { baseIriFromRequest, loader, store } = {}) {
     throw new Error('no loader or store provided')
   }
 
+  if (middleware.resource) {
+    router.use(waitFor(init, () => Router().use(middleware.resource)))
```

Comment:
Why do you wrap it with a `Router().use()`. I think it should work just like this:

```javascript
    router.use(waitFor(init, () => middleware.resource))
```

Reply:
The `middleware.resource` can be an array of middlewares and they will not work while inside `waitFor` the value returned by the factory is directly invoked

Reply:
Will it work, if it's not an array? If anybody wants to use multiple middlewares, it's always possible to wrap it with `express.Router` and hand over the `Router` object to the hydra-box middleware. Adding support for different kinds of input will unnecessarily increase the complexity of the hydra-box code.

Reply:
Yes, why wouldn't it work? There's a test too.

>  it's always possible to wrap it with express.Router and hand over the Router object to the hydra-box middleware

I don't like how this deviates from the way express is used. You can pass a handler or array of handlers to `app.use`/`router.use` and I would expect extension points middleware to behave the same. The is only these to possibilities really.

I think the complexity lies not here but in how the `waitFor` takes a factory function. I would like to simplify that at some point

Reply:
An alternative could be to hide Router inside waitFor

```diff
+const { Router } = require('express')
+
 function waitFor (promise, factory) {
   let middleware = null
 
@@ -5,7 +7,7 @@ function waitFor (promise, factory) {
     await promise
 
     if (!middleware) {
-      middleware = factory()
+      middleware = Router().use(factory())
     }
 
     middleware(req, res, next)
  }
```

Reply:
I looked it up in the [docs](http://expressjs.com/en/4x/api.html#app.use). I was not aware that arrays are also supported, as the argument name suggests something else. Then I would go with the `waitFor` solution.